### DISCUSSION TITLE
How can I show the App title into GNOME 40 top bar and dash panel?

### DISCUSSION THREAD

**Discussion by Dryadxon** [AvaloniaUI/Avalonia#6094] [Q&A] [answered]
Hi everyone!
I'm writing a simple app to learn to use Avalonia for the UI, but I noticed that the program name isn't showed in both the top bar and the dash panel (the dock) of GNOME 40.
I tried to `dotnet run` and `dotnet publish && <PATH TO EXE>/App` but the result didn't change, the way the app runs didn't affect it (At first I thought it was a DEBUG build issue).
I don't know what I'm doing wrong, so I've tested the following apps that use Avalonia and they all don't show their names in the top bar or dock (maybe more, but I couldn't test them all), on my system at least:

- [FVim](https://github.com/yatli/fvim)
- [Core2D](https://github.com/wieslawsoltes/Core2D)
- [AvaloniaILSpy](https://github.com/icsharpcode/AvaloniaILSpy)

I didn't try any other desktop environment, and neither Windows nor macOS, so I don't know if something similar happen somewhere else.
Here some screenshot of what I'm describing, and the code of both `App.axaml` and `MainWindow.axaml`:
![TopPanelTitleMissing](https://user-images.githubusercontent.com/81884588/122418829-06160c00-cf7a-11eb-9280-b4ed157a6759.png)
![Top Panel Title OK](https://user-images.githubusercontent.com/81884588/122418531-ccdd9c00-cf79-11eb-89fb-99776944c7cd.png)
![Dash Panel Title Missing](https://user-images.githubusercontent.com/81884588/122418553-d0712300-cf79-11eb-92ac-8f4f826c45e7.png)
![Dash Panel Title OK](https://user-images.githubusercontent.com/81884588/122418581-d49d4080-cf79-11eb-8e1b-41613d0624a3.png)
![Window Title OK](https://user-images.githubusercontent.com/81884588/122418513-c9e2ab80-cf79-11eb-8b4a-c3a89c2dcbb8.png)

Sorry for the bad English, I hope you understand what I mean.

**Comment by kekekeks** [comment, marked as answer]
You most likely need a well-formed .desktop file with matching StartupWMClass installed either in /usr/share/applications or $HOME/.local/share/applications.

E. g. 
```
[Desktop Entry]
Name=Your Application Name
Exec=/path/to/your/app/executable
Icon=your_app_icon_name
Type=Application
StartupWMClass=YourWmClass
Categories=Utility
```

`your_app_icon_name` is the name of a .png file located in /usr/share/icons or $HOME/.local/share/icons
`YourWmClass` is a string passed to `X11PlatformOptions.WmClass`, which is set to your app assembly name by default (`Assembly.GetEntryAssembly()?.GetName()?.Name`)

**Reply by Dryadxon** [reply]
It worked perfectly with the desktop file, now GNOME shows the correct app name, thank you very much.
